composer/utils/inference.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Inference-related utility functions for model export and optimizations.
+
+Used for exporting models into various formats such ONNX, torchscript etc. and apply optimizations such as fusion.
+"""
+
+import copy
+import logging
+import os
+import tempfile
+from typing import Callable, Optional, Sequence, Union
+
+import torch
+
+from composer.models.base import ComposerModel
+from composer.utils import ObjectStore, dist
+from composer.utils.file_helpers import get_file, is_tar
+from composer.utils.iter_helpers import ensure_tuple
+
+log = logging.getLogger(__name__)
+
+__all__ = [
+    'export_for_inference',
+]
+
+
+def export_for_inference(
+    model: ComposerModel,
+    target_format: str,
+    store_path: str,
+    store_object_store: Optional[ObjectStore] = None,
+    sample_input: Optional[Union[torch.Tensor, Sequence[torch.Tensor]]] = None,
+    surgery_algs: Optional[Union[Callable, Sequence[Callable]]] = None,
```

Comment:
Do you mean this should accept, for example, ``BlurPool()``? If yes, we can't do it here since we don't have ``state`` to pass it to ``apply`` method of algorithms. 

> mention in docstring. 
Will do

Reply:
Left the class form part as is and addressed the other concern.